lib/CertificateTools.js

```diff
@@ -346,38 +339,83 @@ async function resolveCertificate() {
+                if (config.server.passphrase) {
+                    options.passphrase = config.server.passphrase;
+                }
+                if (await fs.exists(keyPath)) {
+                    options.key = await fs.readFile(keyPath);
+                }
+                if (await fs.exists(certPath)) {
+                    options.cert = await fs.readFile(certPath);
+                }
+                if (await fs.exists(pfxPath)) {
+                    options.pfx = await fs.readFile(pfxPath);
+                }
             }
         }
         else {
             // copy certs to local instance
-            ConsoleWriter.info("Copy server certificates from global instance of pbiviz certificate...");
+            ConsoleWriter.info("Copy server certificate from global instance of pbiviz...");
             if (globalPbivizOptions.cert) {
                 await fs.copyFile(globalPbivizOptions.cert, path.join(__dirname, '..', config.server.certificate));
                 options.certificate = config.server.certificate;
             }
             if (globalPbivizOptions.key) {
                 await fs.copyFile(globalPbivizOptions.key, path.join(__dirname, '..', config.server.privateKey));
                 options.privateKey = config.server.privateKey;
-            } 
+            }
             if (globalPbivizOptions.pfx) {
                 await fs.copyFile(globalPbivizOptions.pfx, path.join(__dirname, '..', config.server.pfx));
                 // need to pass whole file instead patho to file
                 options.pfx = await fs.readFile(path.join(__dirname, '..', config.server.pfx));
                 options.passphrase = globalPbivizOptions.passphrase;
-
                 // eslint-disable-next-line require-atomic-updates
                 config.server.passphrase = globalPbivizOptions.passphrase;
-            } 
-
+            }
             await fs.writeFile(path.join(__dirname, confPath), JSON.stringify(config));
         }
     }
     return options;
 }
 
+async function veryfyCertFile(keyPath, certPath, pfxPath, passphrase) {
+    let verifyCertDate;
+    try {
+        let endDateStr;
+        
+        // For Windows OS:
+        if (os.platform() === "win32") {
+            if (!fs.existsSync(pfxPath) || !passphrase) {
+                return false;
+            }
+            let certStr = await exec(`powershell "certutil -p ${passphrase} -dump ${pfxPath}"`);
+            let regex = /(?<=NotAfter: ).*$/gm;
+            endDateStr = regex.exec(certStr);
+
+        }
+        // For Linux and Mac/darwin OS:
+        else if (os.platform() === "linux" || os.platform() === "darwin") {
+            if (!fs.existsSync(certPath)) {
+                return false;
+            }
+            endDateStr = await exec(`openssl x509 -enddate -noout -in ${certPath} | cut -d = -f 2`);
+        }
+
+        let endDate = new Date(Date.parse(endDateStr));
+        verifyCertDate = (endDate - new Date()) > (1000 * 60 * 60 * 24);
```

Comment:
move this number to the top of this file a constant

Reply:
created `certSafeperiod` const;